x-pack/plugins/maps/public/shared/layers/sources/es_source.js

```diff
@@ -78,6 +78,47 @@ export class AbstractESSource extends AbstractVectorSource {
     });
   }
 
+
+  async filterAndFormatPropertiesToHtmlForMetricFields(properties) {
```

Comment:
There are a ton off nested `if` statements inside can we refactor to something more like this to improve readability and control flow?

```
async filterAndFormatPropertiesToHtmlForMetricFields(properties) {
    let indexPattern;
    try {
      indexPattern = await this._getIndexPattern();
    } catch(error) {
      console.warn(`Unable to find Index pattern ${this._descriptor.indexPatternId}, values are not formatted`);
      return properties;
    }

    function formatMetricValue(metricField, propertyValue) {
      if (metricField.type === 'count') {
        return propertyValue;
      }

      const indexPatternField = indexPattern.fields.byName[metricField.field];
      if (!indexPatternField) {
        return propertyValue;
      }

      const htmlConverter = indexPatternField.format.getConverterFor('html');
      return (htmlConverter)
        ? htmlConverter(propertyValue)
        : indexPatternField.format.convert(propertyValue);
    }

    const metricFields = this.getMetricFields();
    const tooltipProps = {};
    metricFields.forEach((metricField) => {
      let value;
      for (const key in properties) {
        if (!properties.hasOwnProperty(key) || metricField.propertyKey !== key) {
          continue;
        }

        value = formatMetricValue(metricField, properties[key]);
        break;
      }
      tooltipProps[metricField.propertyLabel] = (typeof value === 'undefined') ? '-' : value;
    });
    return tooltipProps;
  }
```

Reply:
added small deviation in for-loop body